aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,194 @@
+---
+title: Choose an ASP.NET Core UI
+author: wadepickett
+description: Understand when to use the various ASP.NET Core web UI technologies Microsoft provides and supports.
+ms.author: wpickett
+ms.date: 06/30/2021
+no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: tutorials/choose-web-ui
+---
+# Choose an ASP.NET Core web UI
+
+ASP.NET Core is a complete end to end UI framework. You choose which functionalities to combine that fit your web UI app needs.
+
+## Consider the benefits and costs of server rendered UI and client rendered UI
+
+A helpful place to start is to consider if your web UI app will primarily benefit from pre-rendering UI on the server, or rendering UI at the client browser.
+
+There are three general approaches to building modern web UI with ASP.NET Core:
+
+* Apps that render UI from the server.
+* Apps that render UI on the client in the browser.
+* Hybrid apps that take advantage of both server and client UI rendering approaches. For example, most of the web UI is rendered on the server, and client rendered components are added as needed.
+
+There are benefits and drawbacks to consider when rendering UI on the server or on the client.
+
+### Considering a server rendered UI approach
+
+A web UI app that renders on the server dynamically generates the page's HTML and CSS on the server. The page arrives at the client ready to display.
+
+Benefits:
+
+* The client requirements are minimal since the server does the work of logic and page generation:
+  * Great for low-end devices and low-bandwidth connections.
+  * Allows for a broad range of browser versions at the client.
+  * Quick initial page load times.
+  * Minimal to no code to pull to the client.
```

Comment:
I put "minimal" in there since there is a bit of minimal JavaScript, thinking that was covered that way.  But I could be more explicit.  "Minimal to no JavaScript to pull to the client."

Reply:
@guardrex  - but some folks will be using JavaScript won't they?

Reply:
Yes, but it isn't strictly required ... I mean in terms of a *benefits* list for server-rendered. @wadepickett's latest update looks great to me.

Reply:
All good points, thanks Rick and Luke.

Reply:
@danroth27, does this work for you.  I ended up with "Minimal to no JavaScript to pull to the client."